src/Microsoft.ML.Data/DataView/RowToRowMapperTransform.cs

```diff
@@ -15,13 +15,31 @@
 
 namespace Microsoft.ML.Runtime.Data
 {
+    public interface IColumnInfo
+    {
+        /// <summary>
+        /// The name of a column. This string should always be non-empty.
+        /// </summary>
+        string Name { get; }
+
+        /// <summary>
+        /// The type of the column.
+        /// </summary>
+        ColumnType Type { get; }
+
+        /// <summary>
+        /// The metadata for a column, or <c>null</c> if this column has no metadata.
+        /// </summary>
+        IRow Metadata { get; }
+    }
+
     public sealed class RowMapperColumnInfo
     {
         public readonly string Name;
         public readonly ColumnType ColType;
-        public readonly ColumnMetadataInfo Metadata;
+        public readonly IRow Metadata;
 
-        public RowMapperColumnInfo(string name, ColumnType type, ColumnMetadataInfo metadata)
+        public RowMapperColumnInfo(string name, ColumnType type, IRow metadata)
```

Comment:
>IRow [](start = 65, length = 4)

Any chance you willing to modify one of transforms to show how it's gonna be used?